godtools/App/Share/Data/UserAuthentication/UserAuthentication.swift

```diff
@@ -97,11 +115,56 @@ class UserAuthentication {
         signOutOfAllProviders()
             .map { response in
                 self.lastAuthenticatedProviderCache.deleteLastAuthenticatedProvider()
+                self.mobileContentAuthTokenRepository.deleteCachedAuthToken()
+                
                 return response
             }
             .eraseToAnyPublisher()
     }
     
+    private func authenticateWithMobileContentApi(authProviderResponse: AuthenticationProviderResponse, createUser: Bool) -> AnyPublisher<MobileContentAuthTokenDataModel, Error> {
+        
+        return authProviderResponse.getMobileContentAuthProviderToken().publisher
+            .flatMap({ (providerToken: MobileContentAuthProviderToken) -> AnyPublisher<MobileContentAuthTokenDataModel, Error> in
+                
+                return self.mobileContentAuthTokenRepository.fetchRemoteAuthTokenPublisher(providerToken: providerToken, createUser: createUser)
+                    .mapError { urlResponseError in
+                        return urlResponseError as Error
+                    }
+                    .eraseToAnyPublisher()
+            })
+            .eraseToAnyPublisher()
+    }
+    
+    private func renewAppleTokenPublisher(with provider: AuthenticationProviderInterface) -> AnyPublisher<AuthenticationProviderResponse, Error> {
+        
+        return provider.getAuthUserPublisher()
+            .flatMap { authUserDomainModel -> AnyPublisher<AuthenticationProviderResponse, Error> in
+                
+                let authProviderProfile = AuthenticationProviderProfile(
+                    email: authUserDomainModel?.email,
+                    familyName: authUserDomainModel?.lastName,
+                    givenName: authUserDomainModel?.firstName
+                )
+                
+                let persistedAppleRefreshToken = self.mobileContentAuthTokenRepository.getCachedAuthTokenModel()?.appleRefreshToken
+                
+                let appleAuthProviderResponse = AuthenticationProviderResponse(
+                    accessToken: nil,
+                    appleSignInAuthorizationCode: nil,
+                    idToken: nil,
+                    profile: authProviderProfile,
+                    providerType: .apple,
+                    refreshToken: persistedAppleRefreshToken
+                )
```

Comment:
Hey @levieggertcru, I used `AuthenticationProviderResponse` inside `UserAuthentication` when I fetch the refresh token, so the logic in `AuthenticationProviderResponse+MobileContentAuthProviderToken` will get evaluated.  Would you still prefer to move that logic and avoid doing it this way?

Reply:
Hey @rachaelblue I see what you were able to do.  I didn't see that.

But yeah let's still move that logic since we're doing all the mobile-content-api related things in UserAuthentication due to having the MobileContentAuthTokenRepository reference in there now.

Reply:
Fixed!

Reply:
Thanks @rachaelblue !